### DISCUSSION TITLE
ROS2 Networking with 2 computers

### DISCUSSION THREAD

**Discussion by ValkyrX** [iRobotEducation/create3_docs#423] [Questions]
### How are you connecting to your Create 3?

USB Ethernet (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Ubuntu 22.04

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H 01

### Which RMW is your robot running?

FastRTPS

### Does your robot have an assigned namespace? If so, please share.

no

### Is the robot connected to a network? If so, what is the network type?

Yes wifi network

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

Yes

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

I have managed to get the robot conencted to a pi4 and i can run exples and undock n dock etc, what i now need to do because the robot is ubuntu server there is no desktop so i cant run nav etc, so i have mirrored the install on a ubutnu desktop machine , question is how can i access the ros topics on the robot from the ubuntu desktop , are there any tutorials howto get networking between to ros 2 nodes for the create 3 ? any help would be appreciated 

Thanks
Kurt

**Comment by daryaclark** [comment]
What do you mean by accessing the ROS topics?

**Comment by ValkyrX** [comment]
So if I go on remote pc I don't see any topics when I enter ros 2 topic list

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by kwujciak** [reply]
Just to make sure I understand, you are trying to wireless talk to your robot (not over usb)? Or are you still trying to ssh into the Pi and control it that way just on the desktop?

**Comment by ValkyrX** [comment]
I want to connect to the create 3 from a desktop pc not usb , the robot does have a pi on it for the lidar but I just can't see any topics from the create3

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by kwujciak** [reply]
Is your robot on the same Wi-Fi as your PC? Do they have the same RMW?

**Comment by ValkyrX** [comment]
So , would it matter if the switch on the create is set to usb ? Could that be why ?

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by kwujciak** [reply]
It should be set to usb not bluetooth. Did you cross compare the wifi and RMW?

**Comment by ValkyrX** [comment]
Yes the wifi is the same and also the rmw is the same  , I'm wondering if its multicast issue ?

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by kwujciak** [reply]
Are you on a home or work/university network?

**Comment by ValkyrX** [comment]
It's a work network , not sure if multicast works ? Is there a way to do it without ?

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by kwujciak** [reply]
Yes I would try using Fast DDS with multicast disabled. Look at [this page](https://iroboteducation.github.io/create3_docs/setup/xml-config/#fast-dds) to set it to Fast and disable multicast.

**Comment by ValkyrX** [comment]
Hi, so weirdly I have tried doing the test with the talker and listener on the laptop running ubuntu 22.04 and the pi on the robot and I can see the send and response so multi cast I sworking , I just cant see the create topics, I have set fast rdw , they are both on ros domain 0 .. I cant think of anything else to try ?

Kurt Ingleson


[Title: Twitter - Description: Twitter icon]<https://twitter.com/> [Title: Facebook - Description: Facebook icon] <https://www.facebook.com/>  [Title: LinkedIn - Description: LinkedIn icon] <https://www.linkedin.com/>  [Title: Google Plus - Description: Google Plus icon] <https://plus.google.com/>  [Title: Youtube - Description: Youtube icon] <https://www.youtube.com/>




Did you know ? if you have a support request the fastest way to report it is via our ticketing system available at http://helpdesk.pacloud.online<http://helpdesk.pacloud.online/>

This e-mail and any attached files are confidential and may be legally privileged. If you are not the addressee, any disclosure, reproduction, copying, distribution, or other dissemination or use of this communication is strictly prohibited. If you have received this transmission in error please notify the sender immediately and then delete this mail.
E-mail transmission cannot be guaranteed to be secure or error free as information could be intercepted, corrupted, lost, destroyed, arrive late or incomplete, or contain viruses. The sender therefore does not accept liability for any errors or omissions in the contents of this message which arise as a result of e-mail transmission or changes to transmitted date not specifically approved by the sender.
If this e-mail or attached files contain information which do not relate to our professional activity we do not accept liability for such information
This email has been scanned by the PA Communications Cloud Email Network.

**Comment by ValkyrX** [comment]
Just one more question plz,

How many devices can communicate on 1 ros domain ? I seem to be only able to get 2 to communicate if I add a third it doesn’t work


Kurt Ingleson


[Title: Twitter - Description: Twitter icon]<https://twitter.com/> [Title: Facebook - Description: Facebook icon] <https://www.facebook.com/>  [Title: LinkedIn - Description: LinkedIn icon] <https://www.linkedin.com/>  [Title: Google Plus - Description: Google Plus icon] <https://plus.google.com/>  [Title: Youtube - Description: Youtube icon] <https://www.youtube.com/>




Did you know ? if you have a support request the fastest way to report it is via our ticketing system available at http://helpdesk.pacloud.online<http://helpdesk.pacloud.online/>

This e-mail and any attached files are confidential and may be legally privileged. If you are not the addressee, any disclosure, reproduction, copying, distribution, or other dissemination or use of this communication is strictly prohibited. If you have received this transmission in error please notify the sender immediately and then delete this mail.
E-mail transmission cannot be guaranteed to be secure or error free as information could be intercepted, corrupted, lost, destroyed, arrive late or incomplete, or contain viruses. The sender therefore does not accept liability for any errors or omissions in the contents of this message which arise as a result of e-mail transmission or changes to transmitted date not specifically approved by the sender.
If this e-mail or attached files contain information which do not relate to our professional activity we do not accept liability for such information
This email has been scanned by the PA Communications Cloud Email Network.

**Reply by kwujciak** [reply]
I've been able to put many robots on one domain ID and see all of their topics successfully. In response to the question above, I would imagine that if you cannot see a topic list, multicast could still be the issue if you have it enabled. That would be my best guess. I would also double check that if your Pi is on 22.04, your Create is on the latest Humble firmware (**not** Galactic).

**Comment by ValkyrX** [comment]
They are all on galactic , am I better using humble ?

Thanks

Kurt Ingleson
Engineering Manager
***@***.******@***.***>
Phone number: 0800 0277700<tel:08000277700>
https://pacomms.co.uk<https://pacomms.co.uk/>

[Title: Twitter - Description: Twitter icon]<https://twitter.com/> [Title: Facebook - Description: Facebook icon] <https://www.facebook.com/>  [Title: LinkedIn - Description: LinkedIn icon] <https://www.linkedin.com/>  [Title: Google Plus - Description: Google Plus icon] <https://plus.google.com/>  [Title: Youtube - Description: Youtube icon] <https://www.youtube.com/>
***@***.***
[Title: Hexagon bar - Description: Hexagon bar]
***@***.***<https://pacomms.co.uk/>



Did you know ? if you have a support request the fastest way to report it is via our ticketing system available at http://helpdesk.pacloud.online<http://helpdesk.pacloud.online/>

This e-mail and any attached files are confidential and may be legally privileged. If you are not the addressee, any disclosure, reproduction, copying, distribution, or other dissemination or use of this communication is strictly prohibited. If you have received this transmission in error please notify the sender immediately and then delete this mail.
E-mail transmission cannot be guaranteed to be secure or error free as information could be intercepted, corrupted, lost, destroyed, arrive late or incomplete, or contain viruses. The sender therefore does not accept liability for any errors or omissions in the contents of this message which arise as a result of e-mail transmission or changes to transmitted date not specifically approved by the sender.
If this e-mail or attached files contain information which do not relate to our professional activity we do not accept liability for such information
This email has been scanned by the PA Communications Cloud Email Network.

**Reply by kwujciak** [reply]
Galactic will work fine. In response to putting multiple robots on the same domain ID, I should've mentioned to make sure they all have different namespaces. If multiple robots have different namespaces but same domain ID, you will see a topic list with each one (if you're computer is also on that domain ID). 

Can you tell me the output when you run `echo $RMW_IMPLEMENTATION, $FASTRTPS_DEFAULT_PROFILES_FILE` in terminal?

**Comment by ValkyrX** [comment]
So using galactic I have the topics showing now so maybe it was a humble issue, however if I try and install the examples rosdep fails and says it cant find teleop_twist package and a few others, it didn’t do that on humble ,


Kurt Ingleson






Did you know ? if you have a support request the fastest way to report it is via our ticketing system available at http://helpdesk.pacloud.online<http://helpdesk.pacloud.online/>

This e-mail and any attached files are confidential and may be legally privileged. If you are not the addressee, any disclosure, reproduction, copying, distribution, or other dissemination or use of this communication is strictly prohibited. If you have received this transmission in error please notify the sender immediately and then delete this mail.
E-mail transmission cannot be guaranteed to be secure or error free as information could be intercepted, corrupted, lost, destroyed, arrive late or incomplete, or contain viruses. The sender therefore does not accept liability for any errors or omissions in the contents of this message which arise as a result of e-mail transmission or changes to transmitted date not specifically approved by the sender.
If this e-mail or attached files contain information which do not relate to our professional activity we do not accept liability for such information
This email has been scanned by the PA Communications Cloud Email Network.

**Reply by kwujciak** [reply]
There may be an issue while installing ROS 2. Try running `sudo apt update` and `sudo apt upgrade`, then source your environment by executing `source /opt/ros/galactic/setup.bash`.
